swan/templates/config.yaml

```diff
@@ -6,4 +6,311 @@ metadata:
+
+            if not found:
+                list.append(element)
+
+            return list
+
+    # https://jupyterhub-kubespawner.readthedocs.io/en/latest/spawner.html
+    # This is defined in the configuration to allow overring independently
+    # of which config file is loaded first
+    # c.SwanKubeSpawner.modify_pod_hook = swan_pod_hook
+    def swan_pod_hook(spawner, pod):
+        """
+        :param spawner: Swan Kubernetes Spawner
+        :type spawner: swanspawner.SwanKubeSpawner
+        :param pod: default pod definition set by jupyterhub
+        :type pod: client.V1Pod
+
+        :returns: dynamically customized pod specification for user session
+        :rtype: client.V1Pod
+        """
+        pod_hook_handler = SwanPodHookHandler(spawner, pod)
+        return pod_hook_handler.get_swan_user_pod()
+
+    """
+    Configuration for JupyterHub
+    """
+    c.SwanKubeSpawner.modify_pod_hook = swan_pod_hook
+
+
+    # Hub services
+    # FIXME port is not exposed so it cannot be accessed. Maybe we should run this separately?
+    # if get_config("custom.notificationsService", True):
+    #     c.JupyterHub.services.append(
+    #         {
+    #             'name': 'notifications',
+    #             'command': 'swannotificationsservice --port 8989'.split(),
+    #             'url': 'http://hub:8989'
+    #         }
+    #     )
+
+    # Culling of users and ticket refresh
+    if get_config("custom.cull.enabled", False):
+        cull_cmd = ["swanculler"]
+        base_url = c.JupyterHub.get("base_url", "/")
+        cull_cmd.append("--url=http://localhost:8081" + url_path_join(base_url, "hub/api"))
+
+        cull_timeout = get_config("custom.cull.timeout")
+        if cull_timeout:
+            cull_cmd.append("--timeout=%s" % cull_timeout)
+
+        cull_every = get_config("custom.cull.every")
+        if cull_every:
+            cull_cmd.append("--cull-every=%s" % cull_every)
+
+        if get_config("custom.cull.users"):
+            cull_cmd.append("--cull-users=True")
+
+        if get_config("custom.cull.removeNamedServers"):
+            cull_cmd.append("--remove-named-servers")
+
+        cull_max_age = get_config("custom.cull.maxAge")
+        if cull_max_age:
+            cull_cmd.append("--max-age=%s" % cull_max_age)
+
+        check_eos = get_config('custom.cull.checkEosAuth', False)
+        if not check_eos:
+            cull_cmd.append("--disable-hooks=True")
+
+        hooks_dir = get_config('custom.cull.hooksDir')
+        if hooks_dir:
+            cull_cmd.append(f"--hooks-dir={hooks_dir}")
+
+        c.JupyterHub.services.append(
+            {
+                "name": "cull-idle",
+                "admin": True,
+                "command": cull_cmd,
+            }
+        )
+
+    # init lists for volumes and volume_mounts
+    c.SwanKubeSpawner.volumes = []
+    c.SwanKubeSpawner.volume_mounts = []
+
+    # add EOS to notebook pods
+    {{- if .Values.eosClient.csiDriver }}
```

Comment:
If I understood correctly, this is the only reason why you brought the configs directly to this file.
But can't we just ensure the same result by retrieving the config within the file and doing a normal `if` `else`? (need to check if you can retrieve _any_ value of the config, even those outside of JH).

I would prefer that route, because this makes it inconsistent with the rest of files that get loaded.

Reply:
Also, can't we just make things uniform and use `/eos` or `/var/eos` in both?

Reply:
After a quick check, I think we cannot get these values, unless they are under `hub`, for example.

Reply:
Correct: The file was moved to templates so that be able to render it.
It is on personal preference:
- My option gives k8s a clean, ready-to-use chart according to the choice made and the whole logic is there
-  Your option shifts the choice to Python, which would be equally good.

If my option is inconsistent with other things already implemented (now it is clear to me why you have `/files`), I will move it back and put the logic in python.
To be checked exactly how we can propagate these values to the python script.
Will keep you posted.

Reply:
And for where to mount eos, yes, we can use the same path on the host but I would still prefer to keep the switch for csi VS non-csi. Just in case of future breaking changes and for consistency with cvmfs.

Reply:
Answering to your _I think we cannot get these values, unless they are under hub, or custom_, I think we should put them into `hub.extraEnv`.
How about:
- `swan_config.py` goes back to `/files`
- if/else logic goes to `swan_config.py`
- pass envvars via `hub.extraEnv` to tell python what to do
- define the envvars in `values.yaml` by anchoring the values of {eos, cvmfs}Client.daemonSet/csiDriver
?

Reply:
I slept on this... `extraEnv` is for Environment variables, so not a good place (and I don't want to use env vars in the if/else).
I think it would make sense to put these values under `singleuser`, as these options are used by those pods only (not really in our production, but that's another story). So, something like `singleuser.cvmfs.{}` and `singleuser.eos.{}`. We might want as well to keep only that place as the source of truth and remove all the anchors? This means using that in the Chart dependencies as well. What do you think? (let me just test that those values are available in the Python realm).

I have another observation: we need to add an extra option when selecting the deployment type for EOS/CVMFS.. Why? You can create clusters that come with the CSI components out of the box, so we might want to set both the csi and the daemonset to False. But we still need to add them in the Python config, so in the `if` we should do something like `if csi or forceCsi`.

It's true it's on a personal preference, but if we have this working under `singleuser`, do you like it or not really?

Reply:
> let me just test that those values are available in the Python realm

Looks like it works

Reply:
Yes, you are totally right about the 3rd option. The current solution clearly comes from my biased view of having nothing in the cluster (which is not the case in OpenStack).

Working on the singleuser.cvmfs.{} and singleuser.eos.{}

Reply:
Globals won't work :/
Imlemented the three configuration options (daemonSet, CSI deploy, CSI already there) as per https://github.com/swan-cern/swan-charts/commit/e9e231b93e4130a33afe535cca070a2037526b90